packages/block-library/src/navigation/edit/index.js

```diff
@@ -306,6 +317,23 @@ function Navigation( {
 		);
 	}
 
+	// Show a warning if the selected menu is no longer available.
+	// TODO - the user should be able to select a new one?
+	if ( navigationMenuId && isNavigationMenuMissing ) {
+		return (
+			<div { ...blockProps }>
+				<Warning>
+					{ __(
+						'Navigation menu has been deleted or is unavailable. '
+					) }
+					<Button onClick={ startWithEmptyMenu } variant="link">
+						{ __( 'Create a new menu?' ) }
+					</Button>
+				</Warning>
+			</div>
+		);
+	}
```

Comment:
This was removed in https://github.com/WordPress/gutenberg/pull/36507, I'm not sure why it made a reappearance here.